### PR TITLE
Persist last to redis

### PR DESCRIPTION
**Pull request by captncraig**

### CODE REVIEW

File: cmd/bosun/database/database.go [hunk 1]
```diff
@@ -40,6 +40,9 @@ type DataAccess interface {
 
 	Search_AddMetricTagSet(metric, tagSet string, time int64) error
 	Search_GetMetricTagSets(metric string, tags opentsdb.TagSet) (map[string]int64, error)
+
+	Search_BackupLastInfos(map[string]map[string]*LastInfo) error
```
Comment (kylebrandt) [inline cmd/bosun/database/database.go]: Why underscore here? Should it be a package, method on something, etc? Seems fishy...?

Comment (captncraig) [inline cmd/bosun/database/database.go]: The whole naming is odd. I was trying to segregate the search methods, but there are too many now. I will probably segregate to data.Search.GetWhatever() with a nested interface.

File: cmd/bosun/search/search.go [hunk 2]
```diff
@@ -24,51 +24,52 @@ import (
...
-			s.last[key] = p
+			p = &database.LastInfo{}
+			mmap[dp.Tags.String()] = p
 		}
-		if p.timestamp < dp.Timestamp {
+		if p.Timestamp < dp.Timestamp {
 			if fv, err := getFloat(dp.Value); err == nil {
-				p.diffFromPrev = (fv - p.lastVal) / float64(dp.Timestamp-p.timestamp)
-				p.lastVal = fv
+				p.DiffFromPrev = (fv - p.LastVal) / float64(dp.Timestamp-p.Timestamp)
```
Comment (kylebrandt) [inline cmd/bosun/search/search.go]: dp.Timestamp-p.Timestamp - Check for Div by Zero?

Comment (captncraig) [inline cmd/bosun/search/search.go]: Actually, we already verified that `p.timestamp < dp.Timestamp`. therefore `dp.Timestamp-p.Timestamp` will always be greater than zero.

**Comment by Dieterbe** [comment]
so what exactly does this persist?

**Comment by captncraig** [comment]
Pretty much so it survives restarts. This data is used to serve a host page on opserver, and kyle wanted it  to be available fast. If it gets cleared on restarts, funny things happen. 

I would argue that perhaps that endpoint should be served from opentsdb data, but I think there were good reasons against that. Perhaps @kylebrandt  can refresh me.

**Pull request merged**